sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -214,11 +214,31 @@ class _TestStreamRootBundleProvider(RootBundleProvider):
   """
   def get_root_bundles(self):
     test_stream = self._applied_ptransform.transform
+
+    # The TestStream specification allows for multiple TestStreams in the same
+    # pipeline (with only one controlling the clock). Here, we use an array in
+    # the global EvaluationContext state to keep track of the iterator for each
+    # event stream.
+    idx = len(self._evaluation_context._test_stream_events)
```

Comment:
Is this ever non-empty? It seems repeated calls would append everything. (Or is it populated elsewhere?)

Reply:
In https://github.com/apache/beam/pull/10994 I changed this to a singleton